nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractPutElasticsearch.java

```diff
@@ -158,15 +200,38 @@ void transferFlowFilesOnException(final Exception ex, final Relationship rel, fi
         }
     }
 
-    void logElasticsearchDocumentErrors(final IndexOperationResponse response) throws JsonProcessingException {
-        if (logErrors || getLogger().isDebugEnabled()) {
-            final List<Map<String, Object>> errors = response.getItems();
-            final String output = String.format("An error was encountered while processing bulk operations. Server response below:%n%n%s", errorMapper.writeValueAsString(errors));
+    void handleElasticsearchDocumentErrors(final IndexOperationResponse response, final List<Integer> errorIndices, final ProcessSession session, final FlowFile parent) throws IOException {
+        if (!errorIndices.isEmpty() && (outputErrorResponses || logErrors || getLogger().isDebugEnabled())) {
+            final List<Map<String, Object>> errorResponses = errorIndices.stream().map(index -> response.getItems().get(index)).collect(Collectors.toList());
 
-            if (logErrors) {
-                getLogger().error(output);
-            } else {
-                getLogger().debug(output);
+            if (logErrors || getLogger().isDebugEnabled()) {
+                final String output = String.format(
+                        "An error was encountered while processing bulk operations. Server response below:%n%n%s",
+                        errorMapper.writeValueAsString(errorResponses)
+                );
+
+                if (logErrors) {
+                    getLogger().error(output);
+                } else {
+                    getLogger().debug(output);
+                }
+            }
+
+            if (outputErrorResponses) {
+                FlowFile errorResponsesFF = null;
+                try {
+                    errorResponsesFF = session.create(parent);
+                    try (final OutputStream errorsOutputStream = session.write(errorResponsesFF)) {
+                        errorMapper.writeValue(errorsOutputStream, errorResponses);
+
+                        errorResponsesFF = session.putAttribute(errorResponsesFF, "elasticsearch.put.error.count", String.valueOf(errorResponses.size()));
```

Comment:
Have you tested this live? I've had problems with this in the past where the ProcessSession didn't want to update the attributes while the OutputStream was open.

Reply:
Seemed to work OK when I tried it, but should be able to move this outside of the try-with-resources if you've seen it cause problems in the past

Reply:
I'll test it out locally to verify. I think I had that problem on 1.13, so it may be fixed by now.

Reply:
To be fair, looking at the code it would make more sense for the `transfer` and `attributes` calls to be outside of the try-with-resources, so I've made the change and will push it once all the tests have passed locally 👍